src/moderation.ts

```diff
@@ -259,6 +261,52 @@ export class Moderation<StreamChatGenerics extends ExtendableGenerics = DefaultG
     });
   }
 
+  /**
+   * Experimental: Check user profile
+   *
+   * Warning: This is an experimental feature and the API is subject to change.
+   *
+   * This function is used to check a user profile for moderation.
+   * This will not create any review queue items for the user profile.
+   * You can just use this to check weather to allow certain user profile to be created or not.
+   *
+   * Example:
+   *
+   * ```ts
+   * const res = await client.moderation.checkUserProfile(userId, { username: "fuck_boy_001", profileImage: "https://example.com/profile.jpg" });
+   * if (res.recommended_action === "remove") {
+   *   // Block the user profile from being created
+   * } else {
+   *   // Allow the user profile to be created
+   * }
+   * ```
+   *
+   * @param userId
+   * @param username
+   * @param profileImage
+   * @returns
+   */
+  async checkUserProfile(userId: string, profile: { profileImage?: string; username?: string } = {}) {
+    if (!profile.username && !profile.profileImage) {
+      throw new Error('Either username or profileImage must be provided');
+    }
+
+    return await this.check(
+      MODERATION_ENTITY_TYPES.userprofile,
+      userId,
+      userId,
+      {
+        texts: [profile.username as string],
+        images: [profile.profileImage as string],
```

Comment:
```suggestion
        texts: [profile.username as string],
        images: [profile.profileImage as string],
```
One of these might become: `[undefined]` - is that okay?